python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -993,6 +994,84 @@ void try_except_list_attributes() {
 
   }
 
+  @Test
+  void inferTypeForQualifiedExpression() {
+    var root = inferTypes("""
+      class A:
+        def foo():
+           ...
+      def f():
+        a = A()
+        a.foo()
+      """);
+
+    var qualifiedExpression = TreeUtils.firstChild(root.statements().statements().get(1), QualifiedExpression.class::isInstance)
+      .map(QualifiedExpression.class::cast)
+      .get();
+
+    Assertions.assertThat(qualifiedExpression)
+      .isNotNull()
+      .extracting(QualifiedExpression::typeV2)
+      .isNotNull();
+
+    var qualifiedExpressionType = qualifiedExpression.typeV2();
+    Assertions.assertThat(qualifiedExpressionType)
+      .isInstanceOf(FunctionType.class)
+      .extracting(PythonType::name)
+      .isEqualTo("foo");
+  }
+
+
+  @Test
+  void inferBuiltinsTypeForQualifiedExpression() {
+    var root = inferTypes("""
+      a = [42]
+      a.append(1)
+      """);
+
+    var qualifiedExpression = TreeUtils.firstChild(root.statements().statements().get(1), QualifiedExpression.class::isInstance)
+      .map(QualifiedExpression.class::cast)
+      .get();
+
+    Assertions.assertThat(qualifiedExpression)
+      .isNotNull()
+      .extracting(QualifiedExpression::typeV2)
+      .isNotNull();
+
+    var qualifiedExpressionType = qualifiedExpression.typeV2();
+    Assertions.assertThat(qualifiedExpressionType)
```

Comment:
Here also I think we could have some more meaningful assertions.
the fact that the method is called `append` is one thing. What we want to assert is that this is the `append` of `list`.
Since we don't have FQNs yet in the type model, we can't assert on that. However, maybe we can assert on owner, parameters, etc... Maybe we can have another `list.append` call and assert these are the same types.